code/modules/reagents/chemistry/reagents/impure_reagents.dm

```diff
@@ -16,11 +16,14 @@
 
 /datum/reagent/impurity/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
 	var/obj/item/organ/internal/liver/liver = affected_mob.get_organ_slot(ORGAN_SLOT_LIVER)
+	. = TRUE
 	if(!liver)//Though, lets be safe
 		affected_mob.adjustToxLoss(1 * REM * seconds_per_tick, FALSE, required_biotype = affected_biotype)//Incase of no liver!
-		return ..()
+		..()
+		return .
 	affected_mob.adjustOrganLoss(ORGAN_SLOT_LIVER, liver_damage * REM * seconds_per_tick, required_organ_flag = affected_organ_flags)
-	return ..()
+	..()
+	return .
```

Comment:
Overcomplicated I think, this also doesn't use reagent volume or `current_cycle` so you can just:
```
	. = ..()
	if(isnull(liver)) //Though, lets be safe
		affected_mob.adjustToxLoss(1 * REM * seconds_per_tick, FALSE, required_biotype = affected_biotype)//Incase of no liver!
		return TRUE
	affected_mob.adjustOrganLoss(ORGAN_SLOT_LIVER, liver_damage * REM * seconds_per_tick, required_organ_flag = affected_organ_flags)
	return TRUE
```

Reply:
or even
```
	if(isnull(liver)) //Though, lets be safe
		return affected_mob.adjustToxLoss(1 * REM * seconds_per_tick, FALSE, required_biotype = affected_biotype) //Incase of no liver!
	affected_mob.adjustOrganLoss(ORGAN_SLOT_LIVER, liver_damage * REM * seconds_per_tick, required_organ_flag = affected_organ_flags)
	return TRUE
```